Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/admission/AdmissionScenariosViewModel.kt

```diff
@@ -0,0 +1,61 @@
+package de.rki.coronawarnapp.covidcertificate.person.ui.admission
+
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import de.rki.coronawarnapp.ccl.configuration.update.CCLSettings
+import de.rki.coronawarnapp.ccl.dccadmission.model.storage.DccAdmissionCheckScenariosRepository
+import de.rki.coronawarnapp.ccl.dccwalletinfo.update.DccWalletInfoUpdateTrigger
+import de.rki.coronawarnapp.ccl.ui.text.CCLTextFormatter
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactory
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+
+class AdmissionScenariosViewModel @AssistedInject constructor(
+    format: CCLTextFormatter,
+    private val admissionCheckScenariosRepository: DccAdmissionCheckScenariosRepository,
+    @Assisted private val admissionSharedViewModel: AdmissionSharedViewModel,
+    private val cclSettings: CCLSettings,
+    private val dccWalletInfoUpdateTrigger: DccWalletInfoUpdateTrigger
+) : CWAViewModel() {
+
+    val state = admissionSharedViewModel.admissionScenarios
+        .map { dccCheckScenarios ->
+            State(
+                title = format(dccCheckScenarios.scenarioSelection.titleText),
+                scenarios = dccCheckScenarios.scenarioSelection.items.map { scenario ->
+                    AdmissionItemCard.Item(
+                        identifier = scenario.identifier,
+                        title = format(scenario.titleText),
+                        subtitle = format(scenario.subtitleText),
+                        enabled = scenario.enabled
+                    ) {
+                        selectScenario(scenario.identifier)
+                    }
+                }
+            )
+        }
+        .asLiveData2()
+
+    private fun selectScenario(admissionScenarioId: String) = launch {
+        dccWalletInfoUpdateTrigger.triggerDccWalletInfoUpdateAfterCertificateChange()
```

Comment:
Hmm, the calling the method `triggerDccWalletInfoUpdateAfterCertificateChange()` is a bit confusing since no certificate change actually happened...

Reply:
Yes, this will be addressed by another task